tests/ttnn/integration_tests/resnet/resnet50.py

```diff
@@ -0,0 +1,330 @@
+
+            update_ttnn_module_args(ttnn_module_args.conv1)
+            update_ttnn_module_args(ttnn_module_args.conv2)
+            update_ttnn_module_args(ttnn_module_args.conv3)
+
+            parameters["matmul1"] = preprocess_matmul_alt_conv1_1(conv1_weight, conv1_bias, ttnn_module_args.conv1)
+            parameters["conv2"] = preprocess_conv2d(conv2_weight, conv2_bias, ttnn_module_args.conv2)
+            parameters["matmul3"] = preprocess_matmul_alt_conv1_1(conv3_weight, conv3_bias, ttnn_module_args.conv3)
+
+            if model.downsample is not None:
+                downsample_weight, downsample_bias = fold_batch_norm2d_into_conv2d(
+                    model.downsample[0], model.downsample[1]
+                )
+                update_ttnn_module_args(ttnn_module_args.downsample[0])
+                parameters["downsample"] = preprocess_matmul_alt_conv1_1(
+                    downsample_weight, downsample_bias, ttnn_module_args.downsample[0]
+                )
+        elif isinstance(model, torchvision.models.resnet.ResNet):
+            ttnn_module_args.conv1["activation"] = "relu"  # Fuse relu with conv1
+            conv1_weight, conv1_bias = fold_batch_norm2d_into_conv2d(model.conv1, model.bn1)
+            parameters["conv1"] = fold_conv7s2_into_conv4s1(conv1_weight, conv1_bias, ttnn_module_args.conv1)
+
+            named_parameters = tuple(
+                (name, parameter) for name, parameter in model.named_parameters() if "." not in name
+            )
+            for child_name, child in tuple(model.named_children()) + named_parameters:
+                if child_name in {"conv1", "bn1"}:
+                    continue
+                parameters[child_name] = convert_torch_model_to_ttnn_model(
+                    child,
+                    name=name,
+                    convert_to_ttnn=convert_to_ttnn,
+                    custom_preprocessor=custom_preprocessor,
+                    ttnn_module_args=ttnn_module_args.get(child_name, None),
+                )
+        return parameters
+
+    reader_patterns_cache = {}
+    torch_input_tensor = torch.rand((8, 3, 224, 224), dtype=torch.float32)
+    parameters = preprocess_model(
+        initialize_model=lambda: torch_model,
+        run_model=lambda model: model(torch_input_tensor),
+        reader_patterns_cache=reader_patterns_cache,
+        custom_preprocessor=custom_preprocessor,
+        device=device,
+    )
+
+    torch_model.to(torch.bfloat16)
+    torch_input_tensor_test = torch_input_tensor.to(torch.bfloat16)
+    torch_output_tensor = torch_model(torch_input_tensor_test)
+
+    input_tensor = pad_and_fold_conv_activation_for_unity_stride(
+        torch_input_tensor, *torch_model.conv1.padding, *torch_model.conv1.stride
+    )
+    input_tensor = torch.permute(input_tensor, (0, 2, 3, 1))
+    input_tensor = ttnn.from_torch(input_tensor, dtype=ttnn.bfloat16)
+    output_tensor = parameters.conv1.copy_input_to_device(input_tensor)
```

Comment:
this is pretty nitpicky, but maybe you want to move the model implementation to `ttnn_functional_resnet.py`?

Reply:
Noted.